src/irr/core/CFixedCapacityDoublyLinkedList.h

```diff
@@ -21,20 +21,19 @@ namespace irr {
 		template<typename Value>
 		class DoublyLinkedList
 		{
-		public:
-
+			typedef Snode<Value>* iteratorptr_t;
```

Comment:
make it a `uint32_t`

Reply:
Change the `typedef` to `uint32_t` and cast to `Snode< >` in every function where its used? Isn't that a bit lengthy way to do this?

Reply:
no use the `uint32_t` to index into an array of `SNode` owned by the `CFixedCapacityDoublyLinkedList`.

Reply:
I've had quite a bit of trouble find a way to do it in a performance friendly way, checking online yields only a result similar to my own conclusion. Indexing a linked list will slow things down. Inserting and removing will require a lot more operations, and the action of removing last element and inserting a new one will become `O(2n)`

Reply:
> the action of removing last element and inserting a new one will become O(2n)
actually, just O(n), but still bad 💯